path_login.go

```diff
@@ -139,20 +139,25 @@ func (b *kubeAuthBackend) pathLogin(ctx context.Context, req *logical.Request, d
 		return nil, err
 	}
 
-	aliasName, err := b.getAliasName(role, serviceAccount)
+	aliasName, err := b.getAliasName(role, sa)
 	if err != nil {
 		return nil, err
 	}
 
 	// look up the JWT token in the kubernetes API
-	err = serviceAccount.lookup(ctx, client, jwtStr, role.Audience, b.reviewFactory(config))
-
+	err = sa.lookup(ctx, client, jwtStr, role.Audience, b.reviewFactory(config))
 	if err != nil {
 		b.Logger().Debug(`login unauthorized`, "err", err)
 		return nil, logical.ErrPermissionDenied
 	}
 
-	uid, err := serviceAccount.uid()
+	annotations, err := b.serviceAccountGetterFactory(config).annotations(ctx, client, sa.Namespace, sa.Name)
+	if err != nil {
+		b.Logger().Debug("failed to get service account annotations", "err", err)
```

Comment:
I don't think there's a great need to log the error here as we are returning it to the user - the main reason we log the above error is because we only return an opaque permission denied and otherwise throw the error away. But by getting to here, we know the request can be treated pretty much like an authenticated request and I think it's ok to return the real error to the user.

Reply:
Thank you for the detailed feedback.